src/common/guided_filter.c

```diff
@@ -694,7 +694,7 @@ void guided_filter_cl(int devid, cl_mem guide, cl_mem in, cl_mem out, const int
   assert(w >= 1);
 
   const cl_ulong max_global_mem = dt_opencl_get_max_global_mem(devid);
-  const size_t reserved_memory = (size_t)(dt_conf_get_float("opencl_memory_headroom") * 1024 * 1024);
+  const size_t reserved_memory = (size_t)(dt_opencl_memory_headroom());
```

Comment:
ok, size_t used here, so makes more sense to have dt_opencl_memory_headroom() return an `uint32_t` or a `size_t`.

Reply:
Agreed, will go for a size_t